src/components/AddMovie.tsx

```diff
@@ -1,30 +1,178 @@
 import React from 'react';
 import styled from 'styled-components';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import { MovieData } from '../model';
+import { selectSelectedMovie } from '../store/moviesSlice';
+import { useAppSelector } from '../store/hooks';
+import { GENRES } from './NavBar';
 
 interface AddUpdateMovieProps {
-  onAddMovie?: any;
-  onUpdateMovie?: any;
+  onAddMovie?: (isDataUpdated: boolean, data: MovieData) => void;
+  onUpdateMovie?: (isDataUpdated: boolean, data: MovieData) => void;
 }
 
-const AddUpdateMovieWrapper = styled.div``;
+const AddUpdateMovieWrapper = styled.div`
+  padding: 10px;
+`;
+
+const ItemWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const ErrorMessageWrapper = styled.span`
+  color: red;
+`;
 
 const AddUpdateMovie = (props: AddUpdateMovieProps) => {
```

Comment:
The issue here is that you connect this component to store AND pass some logic(onAddMovie, onUpdateMovie) form parent. This will cause high coupling in future. Remember smart/dumb components? So it's better either to connect the component fully to store and remove props at all, or remove all the logic form this component. My further comments in this file will be regarding removing logic from the component